### ISSUE TITLE
Compile error in custom C# exception that uses map to construct a message

### ISSUE THREAD

**Issue by theJenix** [HaxeFoundation/haxe#11003]
I have the following example custom exception in a file CustomException.hx
```
class CustomException extends cs.system.Exception {

    public function new(parts:Array<String>) {
	super(parts.map(p -> p).join("->"));
    }
}
```

When I try to compile it with

`haxe -cs out CustomException.hx`

I get the following error:

` CustomException.hx:5: lines 5-8 : Types that derive from a native class must have its super() call as the first statement in the constructor`

It looks like it's getting hung up on the map call.  If I remove it and leave the join, it compiles correctly:

```
class CustomException extends cs.system.Exception {

    public function new(parts:Array<String>) {
	super(parts.join("->"));
    }
}
```

I am using Haxe 4.2.5 on a MacBook Pro 14" running Ventura.

Let me know if there's any other information I can provide here.  Thanks!

**Comment by Simn** [comment]
The inlined `map` call is too much for the code transformation to handle this properly. This is partially caused by the analyzer, but the C# code generator doesn't survive it with `@:analyzer(ignore)` either. 

The best you can do here is work around the issue by using a non-inline `map` function. I don't think there will be a more general fix on our end due to various reasons.

**Comment by theJenix** [comment]
Thanks for the response, and I definitely understand.  I dove in a bit more here, not sure if this changes anything on your end but I wanted to document it.

First, I wanted to confirm if this was a C# only issue or if this affected other targets (I think I read something about C# not being a priority in the future, right?).  I tried:
```
class CustomException extends java.lang.Exception {

    public function new(parts:Array<String>) {
	super(parts.map(p -> p).join("->"));
    }
}
```

and `haxe --java out CustomException.hx` and ran into the same issue.  This appears to affect the Java target as well.

I then tried to just implement a loop in an static inline function (which I assume is basically what map is doing, except this is in my code):

```
class CustomException extends java.lang.Exception {

    static inline function transform(parts:Array<String>) {
        var ret = [];
        for (p in parts) {
            ret.push(p);
        }
        return ret;
    }

    public function new(parts:Array<String>) {
	super(transform(parts).join("->"));
    }
}
```

And that gave the same error; removing inline from transform lets it compile, which lines up with what you were saying about inline map.

I then tried to add another method just to :

```
class CustomException extends java.lang.Exception {

    static function transform(parts:Array<String>) {
        var ret = [];
        for (p in parts) {
            ret.push(p);
        }
        return ret;
    }

    static inline function msg(parts:Array<String>) {
        return transform(parts).join("->");
    }

    public function new(parts:Array<String>) {
	super(msg(parts));
    }
}
```

And that failed with the original error; it compiled when I removed inline from msg and I could even put inline back onto transform as long as the call from the constructor was non-inline.

I also tried one where I called an inlined method that returned a constant, and that compiled successfully.

So it looks like the error is caused by calling any inline method that accesses constructor args, from a constructor in a class that extends a native class (even with an intermediate class in between..I tried introducing a base class between CustomException and java.lang.Exception, with the same results).

Is that true?  And, given that, is this expected?  And, is it expected across platforms that are not C#?

Let me know if you'd prefer I open another ticket for Java for this, or if it's appropriate to open a documentation ticket to document this "gotcha".

Thanks!
